External/FEXCore/Source/Interface/IR/Passes/ConstProp.cpp

```diff
@@ -893,6 +904,33 @@ bool ConstProp::ConstantPropagation(IREmitter *IREmit, const IRListView& Current
         IREmit->ReplaceWithConstant(CodeNode, NewConstant);
         Changed = true;
       }
+      else if (SrcIsConstant) {
+        // If the source is constant in a BFI then it can be converted to an or or bic depending.
+        if (ConstantSrc == 0) {
+          // We are wanting to clear the bitfield.
+          IREmit->SetWriteCursor(CodeNode);
+          uint64_t SourceMask = (1ULL << Op->Width) - 1;
+          if (Op->Width == 64)
+            SourceMask = ~0ULL;
```

Comment:
I think this'll trip UBSAN? You can swap it around or use a ternary expression to avoid the shift when `Op->Width == 64`

Reply:
This pattern is actually in a few locations in our source. We probably want to just hammer the idiom all at once if we don't want ubsan to get angry.

Reply:
Refactored ConstProp.cpp to remove this idiom from this file at least. Will need to look around in follow-up PRs to remove this in other files.